check_sophos_xg_disk.pl

```diff
@@ -83,14 +110,47 @@ BEGIN
     default => 0,
 );
 
+$mp->add_arg(
+    spec    => 'verbose',
+    help    => 'Print verbose/debug information',
+    default => 0,
+);
+
 $mp->getopts;
 
-#Open SNMP Session
-my ($session, $error) = Net::SNMP->session(
-    -hostname => $mp->opts->hostname,
-    -version => 'snmpv2c',
-    -community => $mp->opts->community,
-);
+my ($session, $error);
+if (defined($mp->opts->username) && defined($mp->opts->authpassword)) {
+    # SNMPv3 login
+    verb('SNMPv3 login');
+    if (!defined($mp->opts->privpassword)) {
+        verb('SNMPv3 AuthNoPriv login : %s, %s', $mp->opts->username, $mp->opts->authprotocol);
+        ($session, $error) = Net::SNMP->session(
+            -hostname => $mp->opts->hostname,
+            -version => 'snmpv3',
+            -username => $mp->opts->username,
+            -authprotocol => $mp->opts->authprotocol,
+            -authpassword => $mp->opts->authpassword,
+        );
+    } else {
+        verb('SNMPv3 AuthPriv login : %s, %s, %s', ${mp->opts->username}, ${mp->opts->authprotocol}, ${mp->opts->privprotocol});
```

Comment:
You have to remove the curly braces, it should be `$mp->opts->username` instead of `${mp->opts->username}`. And the same for the remaining arguments in the subroutine.